### ISSUE TITLE
Producer.createTopics sync flag ignored

### ISSUE THREAD

**Issue by antjkennedy** [SOHU-Co/kafka-node#518]
When calling `producer.createTopics(['t1'], false, cb)` the async flag is ignored with the method returning immediately

``` javascript
producer.createTopics(topicsList, false, (error, data) => {
    console.log('this should be called before');
    if (error) { console.error(error) };
});
console.log('this');
```

**Comment by hyperlink** [comment]
I don't like this `async` parameter.

It's not clear to me what it's supposed to do. It looks like when `true` it will yields after the first attempt to create the topic  (though it's not really creating a topic just fetching the metadata and letting kafka auto create it) and if `false` it will continue to make attempts and yields until fails or it's successful.

I'm inclined to remove this flag on the next major version update.

**Comment by hyperlink** [comment]
published in `1.0.7`
